### PR TITLE
[eng quality] remove extern crate references

### LINKED ISSUE #3538: [eng quality] remove extern crate references
This is a code quality issue that was brought up in https://github.com/stacks-network/stacks-blockchain/discussions/3412.

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
This change removes unnecessary `extern crate` declarations, as per the changes outlined in https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#an-exception. 

### Applicable issues
- fixes #3538

### Additional info (benefits, drawbacks, caveats)
  - I decided to keep `extern crate` references for some macro imports. Removing these instances would mean macros would need to be individually imported, which is a bit of a hassle (especially for slog macros). For example, removing ` #[macro_use] extern crate serde_derive;` would require me to manually import `Serialize` and `Deserialize` in the files they are used. 

- In `stacks-common/src/deps_common/bech32/[mod.rs](http://mod.rs/)`, I left the import to `alloc`. As the rust docs note, there is an exception to the extern crate rules for sys root crates, which is explained here: https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#an-exception 

- I also decided not to modify any of the files in `benches/`, because at the moment `cargo bench` errors out. This should be addressed in a separate PR.

#### Testing Info 
This build should be tested with the `unix` and `windows` config. 

### Checklist
- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -85,18 +85,27 @@ checksum = "739f4a8db6605981345c5654f3a85b056ce52f37a39d34da03f25bf2151ea16e"
...
+version = "0.7.20"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "1e37cfd5e7657ada45f742d6e99ca5788580b5c529dc78faf11ece6dc702656f"
+checksum = "cc936419f96fa211c1b9166887b38e5e40b19958e5b895be7c1f93adec7071ac"
 dependencies = [
  "memchr",
 ]
 
+[[package]]
+name = "android_system_properties"
```
Comment (jcnelson) [inline Cargo.lock:96]: ???

File: Cargo.lock [hunk 2]
```diff
@@ -309,9 +309,15 @@ checksum = "3441f0f7b02788e948e47f457ca01f1d7e6d92c693bc132c22b087d3141c03ff"
...
 [[package]]
 name = "base64"
-version = "0.13.0"
+version = "0.13.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "904dfeac50f3cdaba28fc6f57fdcddb75f49ed61346676a78c4ffe55877802fd"
+checksum = "9e1b586273c5702936fe7b7d6896644d8be71e6314cfe09d3167c95f712589e8"
+
+[[package]]
+name = "base64"
```
Comment (jcnelson) [inline Cargo.lock:317]: FORBIDDEN DUPLICATE PACKAGE

File: Cargo.lock [hunk 3]
```diff
@@ -770,6 +784,50 @@ dependencies = [
  "zeroize",
 ]
 
+[[package]]
+name = "cxx"
```
Comment (jcnelson) [inline Cargo.lock:788]: What needs `cxx`?

File: Cargo.lock [hunk 4]
```diff
@@ -1213,9 +1279,9 @@ dependencies = [
 
 [[package]]
 name = "httparse"
-version = "1.5.1"
+version = "1.8.0"
```
Comment (jcnelson) [inline Cargo.lock:1282]: We vendored this crate and cleaned it up because it's full of `unsafe`

File: Cargo.lock [hunk 5]
```diff
@@ -1260,25 +1326,48 @@ dependencies = [
...
+ "android_system_properties",
+ "core-foundation-sys",
+ "iana-time-zone-haiku",
+ "js-sys",
+ "wasm-bindgen",
+ "winapi 0.3.9",
+]
+
+[[package]]
+name = "iana-time-zone-haiku"
```
Comment (jcnelson) [inline Cargo.lock:1344]: Pretty sure we'll never support Haiku OS

**Review by jcnelson — COMMENTED** [review]
Is the new Cargo.lock necessary?

File: Cargo.lock (hunk 3 above)
Comment (pavitthrap) [reply Cargo.lock:788]: Tracing it through, it looks like this dependency originates from the "chrono" package being imported as version `0.4.23` as opposed to version `0.4.19`.

**Comment by pavitthrap** [comment]
> Is the new Cargo.lock necessary?

I reverted the changes!

**Review by igorsyl — APPROVED** [review]
FYI I see some failing tests in CI.
Thank you for working on this!
I presume we'll include a git hook to automatically manage imports in a separate PR?

**Comment by pavitthrap** [comment]
> FYI I see some failing tests in CI.
> Thank you for working on this!
> I presume we'll include a git hook to automatically manage imports in a separate PR?

The failing unit test has the error message:`'vm::ast::test::test_cost_tracking_deep_contracts_2_1' has overflowed its stack` 
I'll rebase once develop is merged to next, and that should fix this failure. 

I'm not sure what you mean by "git hook to automatically manage imports" - could you expand on that?

**Comment by igorsyl** [comment]
> > FYI I see some failing tests in CI.
> > Thank you for working on this!
> > I presume we'll include a git hook to automatically manage imports in a separate PR?
> 
> The failing unit test has the error message:`'vm::ast::test::test_cost_tracking_deep_contracts_2_1' has overflowed its stack` I'll rebase once develop is merged to next, and that should fix this failure.
> 
> I'm not sure what you mean by "git hook to automatically manage imports" - could you expand on that?

@pavitthrap https://eugene-babichenko.github.io/blog/2018/11/08/rustfmt-git-hook/

**Review by kantai — APPROVED** [review]

**Pull request merged**